### ISSUE TITLE
stb_image.h requires #define STB_IMAGE_IMPLEMENTATION  ?

### ISSUE THREAD

**Issue by BrianWill** [JoeyDeVries/LearnOpenGL#140]
Compiling in vsc 2017, I had to add this define before the include of stb_image.h,  as described here https://stackoverflow.com/questions/24088002/stb-image-h-in-visual-studio-unresolved-external-symbol

Does it compile fine for you without it? If so, curious.

**Comment by BrianWill** [comment]
For later exercises, I also had to move the define and include for stb_image.h after other includes to avoid conflicts. I'm not sure which includes are causing the conflict.

**Comment by iamRedBell** [comment]
Hi, he uses this file https://github.com/JoeyDeVries/LearnOpenGL/blob/master/src/stb_image.cpp which have that define in it, that's why is working for him.

**Comment by BrianWill** [comment]
Hmm, cmake sets up VS projects pointing to _includes/stb_image.h_, so should the cmake be fixed or should the code? Either way, it's not good for the documented build steps to fail.

**Comment by JoeyDeVries** [comment]
Did you compile the full solution using CMake? 

I've indeed added a .cpp version of `stb_image.h` with the define included and CMake builds that file in a library that's referenced from every sample which then should indeed only require the `#inclue stb_image.h`.

These are the relevant lines in the `CMakeLists.txt` file:
``` add_library(STB_IMAGE "src/stb_image.cpp")```
```set(LIBS ${LIBS} STB_IMAGE) ``` 

If you want to do it manually you indeed want to have both the header and cpp file with the same stb_image.h contents, with the cpp file having the additional `#define STB_IMAGE_IMPLEMENTATION`

**Comment by gameengineer** [comment]
For grins I checked out the latest code, used CMake 3.12.0 to generate a Visual Studio 15 2017 (32bit) solution.  CMake automatically created the STB_IMAGE static library project as prescribed and all 81 projects of the solution built successfully.  I did not try to have CMake create the 64-bit versions of the projects but I imagine it will still succeed.  
I'm posting only to provide a data point on this issue to state that at least for me everything worked fine.
